OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2117,6 +2129,10 @@ private static OSNotificationOpenResult generateOsNotificationOpenResult(JSONArr
    }
 
    private static void fireNotificationOpenedHandler(final OSNotificationOpenResult openedResult) {
+      // TODO: Is there a reason we need the opened handler to be fired from main thread?
+
+      // TODO: Once the NotificationOpenedHandler gets a Worker, we should make sure we add a catch
```

Comment:
All the default behavior is done on a different thread before this call.

The reason we did a catch was so a notification would still be displayed, even though the app is going to crash from an uncaught exception.